registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/RegistrationStatusServiceImpl.java

```diff
@@ -225,6 +237,10 @@ public void updateRegistrationStatus(InternalRegistrationStatusDto registrationS
 				dto.setUpdateDateTime(LocalDateTime.now(ZoneId.of("UTC")));
 				RegistrationStatusEntity entity = convertDtoToEntity(registrationStatusDto, 

```

Comment:
we don't have to remove this line.. bcoz i am handling it by adding this line...
if(updateStatusCode) { entity.setStatusCode(dto.getStatusCode()); }
if we remove that line.. then if in any condition value becomes null... so it will give error of not null constraint exception...

Reply:
oh ok, now I understand. But this will make the code understanding and maintenance tough. Please pass the boolean to the convertDtoToEntity explictly and deal with setting the status code from there